### PR TITLE
feat(errors): support more controller errors

### PR DESCRIPTION
**Pull request by Joshua-Anderson**

### CODE REVIEW

**Comment by Joshua-Anderson** [comment]
@bacongobbler I'm pretty sure I added unit tests for every error.

File: errors.go [hunk 1]
```diff
@@ -13,21 +13,26 @@ import (
...
+	invalidCertMsg       = "Could not load certificate"
+	invalidPodMsg        = "does not exist in application"
+	invalidDomainMsg     = "Hostname does not look valid."
+	invalidVersionMsg    = "version cannot be below 0"
+	invalidKeyMsg        = "Key contains invalid base64 chars"
+	duplicateUserMsg     = "A user with that username already exists."
+	invalidEmailMsg      = "Enter a valid email address."
+	invalidTagMsg        = "No nodes matched the provided labels"
+	duplicateIDMsg       = "App with this id already exists."
+	cancelationFailedMsg = "still has applications assigned. Delete or transfer ownership"
```
Comment (bacongobbler) [inline errors.go]: Proper spelling is "cancellation" :)

**Comment by bacongobbler** [comment]
Once again running on empty. LGTM save for the spelling typo

File: errors.go [hunk 2]
```diff
@@ -13,21 +13,26 @@ import (
...
+	invalidDomainMsg      = "Hostname does not look valid."
+	invalidVersionMsg     = "version cannot be below 0"
+	invalidKeyMsg         = "Key contains invalid base64 chars"
+	duplicateUserMsg      = "A user with that username already exists."
+	invalidEmailMsg       = "Enter a valid email address."
+	invalidTagMsg         = "No nodes matched the provided labels"
+	duplicateIDMsg        = "App with this id already exists."
+	cancellationFailedMsg = "still has applications assigned. Delete or transfer ownership"
+	duplicateDomainMsg    = "Domain is already in use by another application"
+	duplicateKeyMsg       = "Public Key is already in use"
```
Comment (arschles) [inline errors.go]: I'm guessing this really only has a line or two changed. because I'm lazy, can you tell me what actually changed here?

Comment (Joshua-Anderson) [inline errors.go]: Last three lines, L33-35

**Comment by bacongobbler** [comment]
Good to merge. :shipit:

**Pull request merged**